src/index.ts

```diff
@@ -66,14 +44,27 @@ function initialSetup() {
 
   // TODO: validate inputs. Technically the scanner's "max" violation level is 3,
   // where: 1 (high), 2 (moderate), and 3 (low)
-  const inputs = {
+  const inputs: PluginInputs = {
+    reportMode: getInput("report-mode") || "check-runs",
     severityThreshold: parseInt(getInput("severity-threshold")) || 4,
     strictlyEnforcedRules: getInput("strictly-enforced-rules"),
+    deleteResolvedComments: getInput("delete-resolved-comments") === "true",
+    target: getInput("target"),
```

Comment:
This fixes the bug that made it so the diff never got scanned. Later on in this file `scannerFlags.target` always gets updated to be a list of files, either provided by the user or the action. When `filterFindingsToDiffScope` checks to see if the user has input a target, it will always be true.

By separating the input given by the user and the target used by the scanner, we can fix the issue

Reply:
What do you mean "the bug that forced every line to be reported"? Is this an existing issue?

Reply:
Yup, since TS upgrade every line in a file is being pinged instead of just changed files

Reply:
That is interesting.... That is not something that I have experienced. Can you raise an issue with an example of this during the action's run? I would like to take a look and see if I can understand the issue and what is causing it.

Reply:
I was trying to replicate it again on the official release, but it's even more broken for me now and cannot save a second comment anymore on 0.1.8 :(

- Action: https://github.com/mattisdada/simple-lambda-router-ts/actions/runs/4487397056/jobs/7890816581
- PR: https://github.com/mattisdada/simple-lambda-router-ts/pull/4

But you can see it trying to create the erroneous comments in the failures, line 6 shouldn't have a comment applied as it wasn't changed

Reply:
So the true issue here is the order of execution where `updateScannerTarget` is setting the `scannerFlags.target` before we get to `filterFindingsToDiffScope` - at which point the `.target` is already set, so this if check will _never_ evaluate to true, thus we will never continue and _always_ report the finding - correct?

```ts
if (!isInChangedLines(violation, relevantLines) && !scannerFlags.target) {
  continue;
}
```

I wonder if we can change the `updateScannerTarget` function to return a separate `List<String>`, say `filesToScan` without mutating the `scannerFlags.target`.

Reply:
This might also explain some of the odd behavior @jbatt08 was observing in #39

Reply:
You've hit the nail on the head. I'm pretty happy with either approach to be honest.

Reply:
I think that changing the function name from `updateScannerTarget` to `getFilesToScan` and having it return a comma separated list of files without mutation to input flags would be preferable to replicating the `.target` input to both `scannerFlags` and `input`.

We also might want to use `.join(",")` instead of trimming the trailing comma like we are doing today.

Reply:
Looks like you've already made these changes to main at some point, I'll remove the changes